### ISSUE TITLE
`aot.export` segfaults on 0-dim tensor globals

### ISSUE THREAD

**Issue by rkayaith** [iree-org/iree-turbine#1312]
## Summary

`DenseResourceElementsAttr.get_from_buffer` segfaults when `create_tensor_global` tries to materialize a 0-dim (scalar) tensor as an inline initialized global. This is hit whenever a mutable 0-dim buffer is exported, e.g. `BatchNorm2d.num_batches_tracked`.

## Reproducer

```python
import torch
import iree.turbine.aot as aot

class M(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.register_buffer("counter", torch.tensor(0, dtype=torch.int64))

    def forward(self, x):
        self.counter += 1
        return x + self.counter

exported = aot.export(M(), torch.tensor(1, dtype=torch.int64))
```

The crash is at [`ir_utils.py:339`](https://github.com/iree-org/iree-turbine/blob/652bf938/iree/turbine/aot/support/ir_utils.py#L339), where `tensor_type` is `tensor<i64>` (rank-0) and `contents` is an 8-byte buffer from a 0-dim numpy array. Changing the buffer to 1-dim (`torch.tensor([0], ...)`) avoids the crash.

<details>
<summary>Python stack trace</summary>

```
$ python -Xfaulthandler repro.py
```
```
Fatal Python error: Segmentation fault

Current thread 0x00007ff76feca000 (most recent call first):
  File ".../iree/turbine/aot/support/ir_utils.py", line 339 in create_tensor_global
  File ".../iree/turbine/aot/support/procedural/globals.py", line 126 in track
  File ".../iree/turbine/aot/support/procedural/exported_program.py", line 326 in _lift_tensor_to_global
  File ".../iree/turbine/aot/support/procedural/exported_program.py", line 278 in resolve_literal
  File ".../iree/compiler/extras/fx_importer.py", line 2267 in _import_literal
  File ".../iree/compiler/extras/fx_importer.py", line 2301 in _import_input
  File ".../iree/compiler/extras/fx_importer.py", line 1526 in _on_access
  File ".../iree/compiler/extras/fx_importer.py", line 1441 in resolve_node_value
  File ".../iree/compiler/extras/fx_importer.py", line 2203 in _import_argument
  File ".../iree/compiler/extras/fx_importer.py", line 2097 in _import_torch_op_overload
  File ".../iree/compiler/extras/fx_importer.py", line 1595 in import_nodes
  File ".../iree/compiler/extras/fx_importer.py", line 843 in import_program
  File ".../iree/turbine/aot/support/procedural/exported_program.py", line 186 in import_exported_program
  File ".../iree/turbine/aot/compiled_module.py", line 708 in __new__
  File ".../iree/turbine/aot/exporter.py", line 348 in export
  File "repro.py", line 13 in <module>
```

</details>

<details>
<summary>C++ stack trace</summary>

```
#0  0x00007ffe0301be6f in mlir::python::mlir::PyDenseResourceElementsAttribute::getFromBuffer(
        mlir::python::mlir::nb_buffer const&,
        std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&,
        mlir::python::mlir::PyType const&,
        std::optional<unsigned long>, bool,
        mlir::python::mlir::DefaultingPyMlirContext)
      from .../iree/compiler/_mlir_libs/libMLIRPythonSupport-mlir.so
#1  0x00007ffe03031e20 in nanobind::detail::func_create<false, true, ...>(...)::{lambda(...)#1}::__invoke(...)
      from .../iree/compiler/_mlir_libs/libMLIRPythonSupport-mlir.so
#2  0x00007ffea64da782 in ?? () from .../iree/compiler/_mlir_libs/libnanobind-mlir.so
#3  0x000000000054b655 in _PyObject_VectorcallTstate (...) at ../Include/internal/pycore_call.h:92
#4  PyObject_Vectorcall (...) at ../Objects/call.c:299
#5  0x000000000053dc20 in _PyEval_EvalFrameDefault (...) at ../Python/ceval.c:4769
```

</details>

## Environment

- iree-turbine @ 652bf938
- iree-base-compiler 3.10.0rc20260129
- PyTorch 2.10.0
- Python 3.11, Linux
